### PR TITLE
[TEST] test and refactor orderbooks

### PR DESCRIPTION
**Pull request by boxbag**

### CODE REVIEW

File: src/js/ripple/orderbook.js [hunk 1]
```diff
@@ -257,60 +413,72 @@ OrderBook.prototype.incrementOfferCount = function(account) {
...
 
 /**
- * Add offer amount to sum amount being offered by an account
+ * Add amount sum being offered for owner
  *
- * @param {String} account address
- * @param {Object|String} offer amount
- * @return sum
+ * @param {String} account - owner's account address
+ * @param {Object|String} amount - offer amount as native string or JSON currency format
```
Comment (wltsmrz) [inline src/js/ripple/orderbook.js]: IOU format *

File: src/js/ripple/orderbook.js [hunk 2]
```diff
@@ -176,80 +212,200 @@ OrderBook.prototype.unsubscribe = function() {
...
+
+/**
+ * Remove cached owner's funds
  *
- * @param {String} account address
+ * @param {String} account - owner's account address
  */
 
-OrderBook.prototype.removeCachedFunds = function(account) {
+OrderBook.prototype.deleteOwnerFunds = function(account) {
```
Comment (wltsmrz) [inline src/js/ripple/orderbook.js]: I prefer "remove"

File: src/js/ripple/orderbook.js [hunk 3]
```diff
@@ -176,80 +212,200 @@ OrderBook.prototype.unsubscribe = function() {
 };
 
 /**
- * Reset cached owner funds, offer counts
+ * Request orderbook entries from server
+ *
+ * @param {Function} callback
+ */
+
+OrderBook.prototype.requestOffers = function(callback) {
```
Comment (wltsmrz) [inline src/js/ripple/orderbook.js]: Did anything change here or is the diff messed up?

Comment (boxbag) [inline src/js/ripple/orderbook.js]: I moved around a lot of the methods so I think the diff is confused

File: src/js/ripple/orderbook.js (hunk 2 above)
Comment (boxbag) [inline src/js/ripple/orderbook.js]: Do you mind if I ask why?

File: src/js/ripple/orderbook.js [hunk 4]
```diff
@@ -606,72 +661,64 @@ OrderBook.prototype.getBalanceChange = function(node) {
       break;
   }
 
-  result.isValid = !isNaN(result.balance)
-                && UInt160.is_valid(result.account);
+  assert(!isNaN(result.balance) && UInt160.is_valid(result.account), 'node has an invalid balance or account');
```
Comment (wltsmrz) [inline src/js/ripple/orderbook.js]: Can't tell if this exceeds 80 columns

File: test/fixtures/addresses.js [hunk 5]
```diff
@@ -0,0 +1,5 @@
+module.exports.ACCOUNT = 'r9cZA1mLK5R5Am25ArfXFmqgNwjZgnfk59';
+module.exports.OTHER_ACCOUNT = 'rpZc4mVfWUif9CRoHRKKcmhu1nx2xktxBo';
+module.exports.THIRD_ACCOUNT = 'rwBYyfufTzk77zUSKEu4MvixfarC35av1J';
+module.exports.FOURTH_ACCOUNT = 'rJnZ4YHCUsHvQu7R6mZohevKJDHFzVD6Zr';
+module.exports.ISSUER = 'rvYAfWj5gh67oV6fW32ZzP3Aw4Eubs59B';
```
Comment (wltsmrz) [inline test/fixtures/addresses.js]: Let's try to not advertise in tests

File: test/orderbook-test.js [hunk 6]
```diff
@@ -112,123 +114,257 @@ describe('OrderBook', function() {
     book.removeAllListeners('model');
   });
 
-  it('Add cached owner funds', function() {
+  it('Automatic unsubscription with once', function(done) {
```
Comment (wltsmrz) [inline test/orderbook-test.js]: 'Automatic unsubscription when listener removed' probably makes more sense

**Comment by wltsmrz** [comment]
Aside from some naming, LGTM

**Comment by geertweening** [comment]
LGTM

File: src/js/ripple/orderbook.js (hunk 2 above)
Comment (wltsmrz) [inline src/js/ripple/orderbook.js]: "delete" is more technically loaded; there is a `delete` keyword in JS (which isn't used here), etc.

**Pull request merged**